tabbycat/results/result.py

```diff
@@ -519,58 +427,45 @@ def save(self):
-                        debate_team=dt, debate_adjudicator=da, position=pos,
-                        defaults=dict(score=self.get_score(adj, side, pos)))
+                self.ballotsub.teamscorebyadj_set.update_or_create(
+                    debate_team=dt, debate_adjudicator=da,
+                    defaults=self.get_defaults_fields('teamscorebyadj', adj, side))
 
     # --------------------------------------------------------------------------
     # Data setting and retrieval
     # --------------------------------------------------------------------------
 
-    def get_score(self, adjudicator, side, position):
-        try:
-            return self.scoresheets[adjudicator].get_score(side, position)
-        except AttributeError:
-            logger.exception("Tried to get score, but scoresheet doesn't do scores. "
-                    "self.takes_scores is %s", self.takes_scores)
-            return None
-
-    def set_score(self, adjudicator, side, position, score):
-        try:
-            scoresheet = self.scoresheets[adjudicator]
-        except KeyError:
-            logger.exception("Adjudicator %s has a scoresheet but is not assigned \
-                to this debate. This debate's panel has probably been changed since \
-                this ballot was loaded for entry. Best to return to the results page \
-                and re-enter this ballot from scratch.", adjudicator.name)
-        try:
-            scoresheet.set_score(side, position, score)
-        except AttributeError:
-            logger.exception("Tried to set score, but scoresheet doesn't do scores. "
-                    "self.takes_scores is %s", self.takes_scores)
+    def get_winner(self, adjudicator):
+        return next(iter(self.scoresheets[adjudicator].winners()), None)
 
-    def get_declared_winner(self, adjudicator):
-        """Not currently used, in place for future implementation of declared winners."""
-        try:
-            return self.scoresheets[adjudicator].get_declared_winner()
-        except AttributeError:
-            logger.exception("Tried to get declared winner, but scoresheet doesn't do declared winners. "
-                    "self.takes_declared_winners is %s", self.takes_declared_winners)
-            return None
+    def add_winner(self, adjudicator, winner):
+        self.scoresheets[adjudicator].add_declared_winner(winner)
 
-    def set_declared_winner(self, adjudicator, declared_winner):
-        """Not currently used, in place for future implementation of declared winners."""
-        try:
-            return self.scoresheets[adjudicator].set_declared_winner(declared_winner)
-        except AttributeError:
-            logger.exception("Tried to set declared winner, but scoresheet doesn't do declared winners. "
-                    "self.takes_declared_winners is %s", self.takes_declared_winners)
+    def set_winner(self, adjudicator, winners):
+        self.scoresheets[adjudicator].set_declared_winners(winners)
```

Comment:
So, this is a bit confusing, mostly because of the whole BP vs two-team format thing. There are multiple ways to do it, none of which are really satisfying. But it'd be helpful to read if the singular were used whenever it should be a single thing, and the plural were used whenever it should be a list (even if the list contains one item).

On a broader note, because most use-cases have just one winner (with BP non-final elimination rounds being the only exception), I think it's acceptable to break ordinary DRY practice / not always be strict about abstracting "number of winners", in order to be more explicit in our code when exactly one winner is expected. One of the reasons the previous code used `advancing` rather than `winners` was to provide an explicit reminder about when code expected to find two teams. So in this particular case, since this class presupposes two-team ballots, maybe it's acceptable to abstract the "number of winner" mechanics away from `DebateResultByAdjudicator`, and have this class "convert" it to a list of one item before passing it on to the scoresheet?

Reply:
True.

On the broader note, I would still prefer to keep the format in the `ByAdjudicator` for consistency between the various places that use the `DebateResult`.